### PR TITLE
Consolidate unit test CI jobs

### PR DESCRIPTION
**Pull request by piersy**
We had a CI job to run unit-tests and then essentially the same job to run
them with coverage. The running time for these jobs was about equal,
sometimes one would be a bit faster and vice versa.

This PR removes one of the jobs leaving us with just one unit-test job
that tracks coverage.

This has the benefit of halving the chance that a build gets failed due
to a flaky test.

Also standardised the approach to running tests between the remaining
unit test jobs.

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit fa80d56b5e09f099384b37090de4605ab0215a52
<details><summary>coverage:  47.0% of statements across all listed packages</summary><br>
<pre>
coverage:  55.9% of statements in consensus/istanbul
coverage:  46.2% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  45.0% of statements in consensus/istanbul/backend/internal/db
coverage:  28.9% of statements in consensus/istanbul/backend/internal/enodes
coverage:  24.3% of statements in consensus/istanbul/backend/internal/replica
coverage:  68.7% of statements in consensus/istanbul/core
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  75.3% of statements in consensus/istanbul/uptime
coverage: 100.0% of statements in consensus/istanbul/uptime/store
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: dfc875cbce
</details>

File: .circleci/config.yml [hunk 1]
```diff
@@ -112,7 +112,7 @@ jobs:
           paths:
             - geth/<<pipeline.parameters.system-contracts-path>>
 
-  unit-tests:
+  race:
```
Comment (YoduYodu) [inline .circleci/config.yml:115]: I think you forget to add -race?

Comment (YoduYodu) [reply .circleci/config.yml:115]: You might want to just consolidate `unit-tests` and `coverage`, and leave `race` as it is?

File: .circleci/config.yml [hunk 2]
```diff
@@ -122,7 +122,7 @@ jobs:
             - go-mod-v1-{{ checksum "go.sum" }}
       - run: go get github.com/jstemmer/go-junit-report
       - run:
-          name: Run Tests
+          name: Run tests
           command: |
             mkdir -p /tmp/test-results
             trap "go-junit-report < /tmp/test-results/go-test.out > /tmp/test-results/go-test-report.xml" EXIT
```
Comment (mcortesi) [inline .circleci/config.yml:128]: agree with @YoduYodu, next line is missing the `-race` parameter

**Review by mcortesi — CHANGES_REQUESTED** [review]
add -race and we're done

File: .circleci/config.yml (hunk 1 above)
Comment (piersy) [reply .circleci/config.yml:115]: So I guess I dropped the race flag by accident when I standardised the approach to running tests, probably copy and pasted the block from unit-tests to race.

**Review by YoduYodu — APPROVED** [review]
LGTM

**Review by mcortesi — APPROVED** [review]

**Pull request merged**